plugin/viewdoc_info.vim

```diff
@@ -204,16 +201,19 @@ endfunction
 " node names from manual, whose name is param1 when invoked for any other
 " parameter
 function s:CompleteInfo(ArgLead, CmdLine, CursorPos)
-	let parts = split(strpart(a:CmdLine, 0, a:CursorPos).'|')
-	if len(parts)>2
-		let heads = system(g:viewdoc_info_cmd . ' --subnodes ' .
-		\           shellescape(parts[1], 1) . " 2>/dev/null | grep '^File: .*,  Node:'")
-		return substitute(heads, 'File: [^\n]*,  Node: \([^,]*\),  [^\n]*', '\1', 'g')
-	else
-		return substitute(substitute(globpath(g:viewdoc_info_path, '*.info*'),
-		\                            '[^\n]*/\([^/]\+\).info[^\n]*', '\1', 'g'),
-		\                 '\([^\n]*\n\)\1*', '\1', 'g')
-  endif
+	let part = join(split(substitute(a:CmdLine[0:a:CursorPos], '\\', '', 'g'))[1:], ' ')
+	let base_cmd = g:viewdoc_info_cmd . " '(dir)Top' 2>/dev/null"
+	let keys_pipe = ' | sed -n ''s/\* \([^:]*\): (.*/\1/p'''
+	let ext_pipe = ' | sed -n ''s/\* [^:]*: \(([^.]*\)\..*/\1/p'' | sort | uniq'
+	if len(part) == 0
+		return system(base_cmd . keys_pipe)
+	endif
+	let pipe = keys_pipe
+	if part[0] == '('
+		let pipe = ext_pipe
+	endif
+	let part = part . (len(a:ArgLead) == 0 ? ' ' : '')
+	return escape(system(base_cmd . pipe . " | sed -n '/^" . part . "/p'"), ' ')
```

Comment:
Here, `part` is inside single quotes, so shellescape won't make sense. Should I remove all single quotes from `part`?